src/tools.f90

```diff
@@ -365,10 +371,19 @@ subroutine restart(ux1,uy1,uz1,dux1,duy1,duz1,ep1,pp3,phi1,dphi1,px1,py1,pz1,rho
           write(111,fmt2) 'iscalar=  ',iscalar
           write(111,fmt2) 'numscalar=',numscalar
           write(111,'(A,I14)') 'itimescheme=',itimescheme
-          write(111,fmt2) 'iimplicit=',iimplicit
+          write(111,'(A,I16)') 'iimplicit=',iimplicit
           write(111,'(A)')'/End'
           write(111,'(A)')'!========================='
 
+          if(particle_active) then
+            write(111,'(A)')'&ParTrack'
+            write(111,'(A)')'!========================='
+            write(111,'(A,I13)') 'n_particles=  ',n_particles
+            write(111,'(A)')     'particle_res_file=  "checkpoint-particles"'
```

Comment:
For a normal restart, the checkpoint file for particle is "checkpoint-particles"
The line is to keep "", otherwise it would be a error when reading restart.